### PR TITLE
Add support for pandoc 2.11

### PR DESCRIPTION
**Pull request by dhimmel**
refs https://github.com/manubot/rootstock/pull/386

- CI: upgrade pandoc to 2.11.1.1 where possible
- GH actions: build on multiple pandoc versions
- GH actions: install compatible panflute version with pandoc version
- manubot.pandoc.bibliography.load_bibliography: pandoc 2.11 support
  Converting nbib and ris formats to CSL JSON requires pandoc-citeproc,
  which is no longer maintained or packaged with Pandoc >= 2.11.
  Keep support if pandoc-citeproc is available on system.

### CODE REVIEW

**Comment by dhimmel** [comment]
Regarding pandoc 2.11.1.1 installation on Windows issue in the GitHub actions, reported upstream at https://github.com/r-lib/actions/issues/206.

**Comment by dhimmel** [comment]
@agitter at a point where I think this is mergeable if you want to take a look (no obligation).

The pandoc 2.11 is a bit disruptive with pandoc-citeproc going away.

File: manubot/pandoc/tests/test_bibliography.py [hunk 1]
```diff
@@ -6,48 +6,47 @@
...
-    directory / x for x in directory.glob("bibliographies/bibliography.*")
+
+
+skipif_no_pdcp = pytest.mark.skipif(
+    not shutil.which("pandoc-citeproc"),
+    reason="pandoc-citeproc required to load this format.",
 )
-bibliography_path_ids = [path.name for path in bibliography_paths]
+skipif_no_pd = pytest.mark.skipif(
+    not shutil.which("pandoc-citeproc"), reason="pandoc required to load this format."
```
Comment (agitter) [inline manubot/pandoc/tests/test_bibliography.py:16]: What's this test condition?  Should it skip if pandoc is missing?

**Review by agitter — COMMENTED** [review]
Looks good to me except for the one line I commented on.

File: manubot/pandoc/tests/test_bibliography.py (hunk 1 above)
Comment (dhimmel) [reply manubot/pandoc/tests/test_bibliography.py]: yes! great catch. fixed in https://github.com/manubot/manubot/pull/256/commits/b1f7363b2f405262b8965b1a631b4e66ceaa0bd0

**Comment by dhimmel** [comment]
encountered an upstream issue causing the `CiteprocLocaleNotFound: C.UTF-8` error on GitHub Actions ubuntu builds for pandoc 2.11.1.1. Reported at https://github.com/jgm/citeproc/issues/26

**Pull request merged**